### PR TITLE
#4221 follow up end date api

### PR DESCRIPTION
**Pull request by ssferrazza**
Bugfix for #4221

### CODE REVIEW

File: sormas-backend/src/main/java/de/symeda/sormas/backend/person/PersonFacadeEjb.java [hunk 1]
```diff
@@ -379,7 +379,10 @@ public void validate(PersonDto source) throws ValidationRuntimeException {
 			cq.where(filter);
 		}
 
-		cq.multiselect(personJoin.get(Person.UUID), contactRoot.get(Contact.FOLLOW_UP_UNTIL));
+		final Expression<Object> followUpStatusExpression = cb.selectCase()
```
Comment (FredrikSchaeferVitagroup) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/person/PersonFacadeEjb.java:382]: There also is a getCaseLatestFollowUpEndDates(...) in this class, which needs this fix, too.

**Review by FredrikSchaeferVitagroup — CHANGES_REQUESTED** [review]

**Comment by FredrikSchaeferVitagroup** [comment]
When trying to call `localhost:6080/sormas-rest/visits-external/followUpEndDates/0` with case follow up enabled, I am getting:

```
Caused by: java.lang.NullPointerException
	at java.base/java.util.Comparator.lambda$comparing$77a9974f$1(Comparator.java:469)
	at java.base/java.util.function.BinaryOperator.lambda$maxBy$1(BinaryOperator.java:74)
	at java.base/java.util.stream.Collectors$1OptionalBox.accept(Collectors.java:918)
	at java.base/java.util.stream.Collectors.lambda$groupingBy$53(Collectors.java:1129)
	at java.base/java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
	at java.base/java.util.stream.DistinctOps$1$2.accept(DistinctOps.java:175)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1654)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.forEachRemaining(StreamSpliterators.java:312)
	at java.base/java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:735)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at de.symeda.sormas.backend.person.PersonFacadeEjb.getLatestFollowUpEndDates(PersonFacadeEjb.java:350)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)

```

**Comment by FredrikSchaeferVitagroup** [comment]
I am now having another problem: I have a person with two contacts. One has the follow up cancelled, the other has not. When I make the API call, I get `"latestFollowUpEndDate": null` for that person. This is not correct, as the person is under follow up.
Please also consider that in scenarios where a person has a contact with cancelled follow up and a case with ongoing follow up.

**Comment by ssferrazza** [comment]
> I am now having another problem: I have a person with two contacts. One has the follow up cancelled, the other has not. When I make the API call, I get `"latestFollowUpEndDate": null` for that person. This is not correct, as the person is under follow up.
> Please also consider that in scenarios where a person has a contact with cancelled follow up and a case with ongoing follow up.

This was caused by the use of `.stream().distinct()` and the sorting/order of the result set. Postgres by default returns nulls first and therefore a call to `.stream().distinct()` considering the `equals()` method of the `PersonFollowUpEndDto` returned the wrong entry. The bugfix was to return the result set with nulls last using a `coalesce` in the criteria api.

**Review by FredrikSchaeferVitagroup — APPROVED** [review]

**Pull request merged**